### PR TITLE
refactor: Add tracking performance as function of DeltaR

### PR DESCRIPTION
**Pull request by Corentin-Allaire**
This PR add an extra histogram to the performance monitoring showing the tracking efficiency as function of the distance to the closest nearby different truth particle (as used in the Track ML paper : https://doi.org/10.1007/s41781-023-00094-w)

### CODE REVIEW

**Review by asalzburger — DISMISSED** [review]
Happy with that!

**Comment by paulgessinger** [comment]
@Corentin-Allaire Can you make the PR title a bit more descriptive?

**Comment by Corentin-Allaire** [comment]
> @Corentin-Allaire Can you make the PR title a bit more descriptive?

Done

**Comment by paulgessinger** [comment]
I think the references need an update as well.

**Comment by Corentin-Allaire** [comment]
> I think the references need an update as well.

Yes I will download the artefact and update the ref

**Comment by Corentin-Allaire** [comment]
> > I think the references need an update as well.
> 
> Yes I will download the artefact and update the ref

@paulgessinger I updated the ref so now I should pass but the plots are not printed in the physmon. I guess I need to update the plot making somewhere but if you could point me in the right direction that would be helpful !

**Comment by paulgessinger** [comment]
@Corentin-Allaire That's weird. It seems to run the actual comparison but then fails to build the output plot. I'd need to check the job log or run the plotting locally to figure out what's happening.

**Comment by Corentin-Allaire** [comment]
> @Corentin-Allaire That's weird. It seems to run the actual comparison but then fails to build the output plot. I'd need to check the job log or run the plotting locally to figure out what's happening.

@paulgessinger let me know next week if there is any way I can help !

**Comment by paulgessinger** [comment]
@Corentin-Allaire It chokes on the label:

```
2023-11-17T18:08:35.2321809Z \DeltaR
2023-11-17T18:08:35.2324783Z ^
2023-11-17T18:08:35.2326304Z ParseFatalException: Unknown symbol: \DeltaR, found '\'  (at char 0), (line:1, 
2023-11-17T18:08:35.2327461Z col:1)
```

File: Examples/Framework/include/ActsExamples/Validation/EffPlotTool.hpp [hunk 1]
```diff
@@ -33,14 +33,18 @@ class EffPlotTool {
     std::map<std::string, PlotHelpers::Binning> varBinning = {
         {"Eta", PlotHelpers::Binning("#eta", 40, -4, 4)},
         {"Phi", PlotHelpers::Binning("#phi", 100, -3.15, 3.15)},
-        {"Pt", PlotHelpers::Binning("pT [GeV/c]", 40, 0, 100)}};
+        {"Pt", PlotHelpers::Binning("pT [GeV/c]", 40, 0, 100)},
+        {"DeltaR", PlotHelpers::Binning("DeltaR", 100, 0, 0.3)}};
```
Comment (paulgessinger) [inline Examples/Framework/include/ActsExamples/Validation/EffPlotTool.hpp:37]:
```suggestion
        {"DeltaR", PlotHelpers::Binning("#Delta R", 100, 0, 0.3)}};
```

File: Examples/Framework/src/Validation/EffPlotTool.cpp [hunk 2]
```diff
@@ -36,12 +37,17 @@ void ActsExamples::EffPlotTool::book(
   // efficiency vs phi
   effPlotCache.trackEff_vs_phi = PlotHelpers::bookEff(
       "trackeff_vs_phi", "Tracking efficiency;Truth #phi;Efficiency", bPhi);
+  // efficiancy vs distance to the closest truth particle
+  effPlotCache.trackEff_vs_DeltaR = PlotHelpers::bookEff(
+      "trackeff_vs_DeltaR",
+      "Tracking efficiency;Closest track #DeltaR;Efficiency", bDeltaR);
```
Comment (paulgessinger) [inline Examples/Framework/src/Validation/EffPlotTool.cpp:43]:
```suggestion
      "Tracking efficiency;Closest track #Delta R;Efficiency", bDeltaR);
```

**Comment by Corentin-Allaire** [comment]
Maybe we should completely remove the #Delta ? It looks better but if it crash the CI it is not worth it

**Comment by paulgessinger** [comment]
@Corentin-Allaire I think the plot is there now: 

image

The GitHub-side CI just didn't run because of conflicts in one file. Can you resolve those?

**Comment by Corentin-Allaire** [comment]
> @Corentin-Allaire I think the plot is there now:
> image (The label still doesn't look great, but that's fine for now I'd say)
> 
> The GitHub-side CI just didn't run because of conflicts in one file. Can you resolve those?

@paulgessinger sure ! I can still fetch the artefact to update the physmon right ?

**Comment by paulgessinger** [comment]
@Corentin-Allaire You might be able to fetch it from the GitLab CI, but it might not match exactly (different environment). Otherwise, you'd have to just resolve the conflict by going back to the reference file from `main`, let the CI run, and then update it again.

**Comment by Corentin-Allaire** [comment]
@paulgessinger, now that it pass the tests could you add the auto-merge tag to this ?

**Review by paulgessinger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :white_check_mark: Athena integration test results [d6b8266020e55605d30709e2aa5019ce77ce4cf9]
### :white_check_mark: All tests successful

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174182) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsBenchmarkWithSpot.sh 8 100](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174181) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174180) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateAmbiguityResolution.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174179) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateResolvedTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174178) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174177) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174176) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174175) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174174) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174173) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174172) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174171) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsGSFRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174170) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174169) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174168) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174167) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174166) |
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174165) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174164) |
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/34174163) |

**Pull request merged**